### PR TITLE
feat: adjust HttpServer to HTTPServer and fix err.Error() panic

### PR DESCRIPTION
**Pull request by daheige**
1. adjust httpserver.HttpSever to HTTPServer
2. fix err.Error() panic

### CODE REVIEW

File: apiserver/httpserver/client_access.go [hunk 1]
```diff
@@ -63,91 +63,103 @@ func (h *Httpserver) GetClientAccessServer(include []string) (*restful.WebServic
...
-func (h *Httpserver) ReportClient(req *restful.Request, rsp *restful.Response) {
+func (h *HTTPServer) ReportClient(req *restful.Request, rsp *restful.Response) {
 	handler := &Handler{req, rsp}
-
 	client := &api.Client{}
 	context, err := handler.Parse(client)
 	if context == nil {
-		ret := api.NewResponseWithMsg(api.ParseException, err.Error())
-		handler.WriteHeaderAndProto(ret)
+		var errMsg string
```
Comment (chenhengqi) [inline apiserver/httpserver/client_access.go:94]: This makes no sense. Error check go first.

**Comment by daheige** [comment]
if err == nil ,this will panic.
Unless you have capture processing upstream, this is the error processing of the component library. Keep it rigorous instead of getting it directly.


 原始邮件 
发件人: Hengqi ***@***.***>
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
发送时间: 2021年9月8日(周三) 23:46
主题: Re: [polarismesh/polaris] feat: adjust HttpServer to HTTPServer andfix err.Error() panic (#31)


@chenhengqi commented on this pull request.
In apiserver/httpserver/client_access.go:
> client := &api.Client{} context, err := handler.Parse(client) if context == nil { - ret := api.NewResponseWithMsg(api.ParseException, err.Error()) - handler.WriteHeaderAndProto(ret) + var errMsg string
This makes no sense. Error check go first.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by daheige** [comment]
Error err should be judged in advance instead a context= Nil, it seems that it is not professional at all, not authentic code. Another is that context is a keyword. Please don't use this as a variable name. You can use CTX to name it. Isn't it fragrant? Another point is that the whole project code has not done some good code formatting. Standard packages and third-party packages need to be separated. Looking at many, here and there, it is very unprofessional. Is this the style of big factories?&nbsp;


Can you be more professional and format the code instead of standard packages, and put some third-party packages here and there. A lot of code comments look uncomfortable. It is understandable to use tools to generate them, but they should also comply with the go code specification. Let's not talk about the quality first. It is suggested that the code specification and format should be done well, and then open source. At present, open source is really unprofessional.




------------------ Original ------------------
From: Hengqi Chen ***@***.***&gt;
Date: Wed,Sep 8,2021 11:46 PM
To: polarismesh/polaris ***@***.***&gt;
Cc: heige ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [polarismesh/polaris] feat: adjust HttpServer to HTTPServer and fix err.Error() panic (#31)





 
@chenhengqi commented on this pull request.
 
 
In apiserver/httpserver/client_access.go:
 &gt;  client := &amp;api.Client{}  context, err := handler.Parse(client)  if context == nil { -ret := api.NewResponseWithMsg(api.ParseException, err.Error()) -handler.WriteHeaderAndProto(ret) +var errMsg string  
This makes no sense. Error check go first.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by daheige** [comment]
context rename to ctx，is better ，are you ok?



------------------ Original ------------------
From: Hengqi Chen ***@***.***&gt;
Date: Wed,Sep 8,2021 11:46 PM
To: polarismesh/polaris ***@***.***&gt;
Cc: heige ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [polarismesh/polaris] feat: adjust HttpServer to HTTPServer and fix err.Error() panic (#31)





 
@chenhengqi commented on this pull request.
 
 
In apiserver/httpserver/client_access.go:
 &gt;  client := &amp;api.Client{}  context, err := handler.Parse(client)  if context == nil { -ret := api.NewResponseWithMsg(api.ParseException, err.Error()) -handler.WriteHeaderAndProto(ret) +var errMsg string  
This makes no sense. Error check go first.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by andrewshan** [comment]
> context rename to ctx，is better ，are you ok?
> […](#)
> ------------------ Original ------------------ From: Hengqi Chen ***@***.***&gt; Date: Wed,Sep 8,2021 11:46 PM To: polarismesh/polaris ***@***.***&gt; Cc: heige ***@***.***&gt;, Author ***@***.***&gt; Subject: Re: [polarismesh/polaris] feat: adjust HttpServer to HTTPServer and fix err.Error() panic (#31) @chenhengqi commented on this pull request. In apiserver/httpserver/client_access.go: &gt; client := &amp;api.Client{} context, err := handler.Parse(client) if context == nil { -ret := api.NewResponseWithMsg(api.ParseException, err.Error()) -handler.WriteHeaderAndProto(ret) +var errMsg string This makes no sense. Error check go first. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by andrewshan** [comment]
> context rename to ctx，is better ，are you ok?
> […](#)
> ------------------ Original ------------------ From: Hengqi Chen ***@***.***&gt; Date: Wed,Sep 8,2021 11:46 PM To: polarismesh/polaris ***@***.***&gt; Cc: heige ***@***.***&gt;, Author ***@***.***&gt; Subject: Re: [polarismesh/polaris] feat: adjust HttpServer to HTTPServer and fix err.Error() panic (#31) @chenhengqi commented on this pull request. In apiserver/httpserver/client_access.go: &gt; client := &amp;api.Client{} context, err := handler.Parse(client) if context == nil { -ret := api.NewResponseWithMsg(api.ParseException, err.Error()) -handler.WriteHeaderAndProto(ret) +var errMsg string This makes no sense. Error check go first. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

----
yeah, it suggests to fix it as that:
1. context rename to ctx
2. err check go first, if err != nil, nested logic shouldn't continue

**Comment by daheige** [comment]
@andrewshan Ok i have adjusted

**Pull request merged**